src/hotspot/share/memory/allocation.hpp

```diff
@@ -494,6 +494,10 @@ class AnyObj {
 #endif // ASSERT
 
  public:
+  void* operator new(size_t size, void* ptr) {
+    return ptr;
+  }
+
```

Comment:
This should probably be a separate RFE.

Reply:
I think this change should not be made.  Callers should be using global placement new.  That's what we
do everywhere else this comes up.  (It has always seemed like a bug to me that the operation "global
placement new" is syntactically an allocation function, and so subject to this kind of name lookup
collision, when it doesn't allocate.)

Reply:
Hi,

100% agree. To be frank, I simply forgot that we had access to the global placement new through prepending `::`.